warehouse/router/state_update_table_uploads.go

```diff
@@ -0,0 +1,19 @@
+package router
+
+import "fmt"
+
+func (job *UploadJob) updateTableUploadsCounts() error {
+	// TODO: THis needs to happen within a transaction
```

Comment:
Are we saying that all the populate inside the below for loop need to happen within the same tx? If yes it's not complex, we can create task for it.

Reply:
I added in this: https://github.com/rudderlabs/rudder-server/pull/4099

Reply:
Will remove the comment.